client/ayon_core/pipeline/workfile/utils.py

```diff
@@ -124,3 +142,624 @@ def should_open_workfiles_tool_on_launch(
     if output is None:
         return default_output
     return output
+
+
+def _get_event_context_data(
+    project_name: str,
+    folder_entity: dict[str, Any],
+    task_entity: dict[str, Any],
+    host_name: str,
+):
+    return {
+        "project_name": project_name,
+        "folder_id": folder_entity["id"],
+        "folder_path": folder_entity["path"],
+        "task_id": task_entity["id"],
+        "task_name": task_entity["name"],
+        "host_name": host_name,
+    }
+
+
+def save_workfile_info(
+    project_name: str,
+    task_id: str,
+    rootless_path: str,
+    host_name: str,
+    version: Optional[int],
+    comment: Optional[str],
+    description: Optional[str],
+    username: Optional[str] = None,
+    workfile_entities: Optional[list[dict[str, Any]]] = None,
+):
```

Comment:
This could do with docstring and also "dataclassed" workfile entity that it returns :)

```suggestio
def save_workfile_info(
    project_name: str,
    task_id: str,
    rootless_path: str,
    host_name: str,
    version: Optional[int],
    comment: Optional[str],
    description: Optional[str],
    username: Optional[str] = None,
    workfile_entities: Optional[list[dict[str, Any]]] = None,
) -> dict[str, Any]:
```